### ISSUE TITLE
Upload logs and artifacts even when task is cancelled

### ISSUE THREAD

**Issue by benoit74** [openzim/zimfarm#965]
Currently, when a Zimfarm task is cancelled (periodic check or manual), logs and artifacts are not uploaded.

We need to upload both of them even if the task is cancelled, since we usually need them for investigation.

There are probably edge-cases where the scraper container will be gone or in such a bad shape that it won't be possible, but we should give our best at trying to do it whenever possible.

**Comment by elfkuzco** [comment]
The worker manager sends a stop signal via `stop_task_worker` which internally calls `docker.api.stop` with a timeout of 20 seconds. According to the [docker docs](https://docs.docker.com/reference/cli/docker/container/stop/), if the container does not exit after the timeout elapses, it is forcibly killed with a `SIGKILL` signal. And if we set it to -1, the daemon waits indefinitely for the container to exit. 

Based on this, should we set the timeout to -1 and make the task worker reliably kill itself even if something goes rogue?

**Comment by rgaudin** [comment]
No, most of the time, 20s is enough (we could increase if we know how long is actually required) but we kill it because it's not going to shutdown properly. This happens and will happen.
